packages/swapper/src/utils.ts

```diff
@@ -30,3 +44,150 @@ export const createTradeAmountTooSmallErr = (details?: {
+}
+declare var Proxy: ProxyConstructor
+
+export const makeSwapperAxiosServiceMonadic = (service: AxiosInstance, _swapperName: SwapperName) =>
+  new Proxy<
+    AxiosInstance,
+    {
+      get: <T = any>(
+        url: string,
+        config?: AxiosRequestConfig<any>,
+      ) => Promise<Result<AxiosResponse<T>, SwapErrorRight>>
+      post: <T = any>(
+        url: string,
+        data: any,
+        config?: AxiosRequestConfig<any>,
+      ) => Promise<Result<AxiosResponse<T>, SwapErrorRight>>
+    }
+  >(service, {
+    get: (trappedAxios, method: 'get' | 'post') => {
+      const originalMethodPromise = trappedAxios[method]
+      return async (...args: [url: string, dataOrConfig?: any, dataOrConfig?: any]) => {
+        // getMixPanel()?.track(MixPanelEvent.SwapperApiRequest, {
+        //   swapper: swapperName,
+        //   url: args[0],
+        //   method,
+        // })
+        const result = await AsyncResultOf(originalMethodPromise(...args))
+
+        return result
+          .mapErr(e =>
+            makeSwapErrorRight({
+              message: 'makeSwapperAxiosServiceMonadic',
+              cause: e,
+              code: TradeQuoteError.QueryFailed,
+            }),
+          )
+          .andThen<AxiosResponse>(result => {
+            if (!result.data)
+              return Err(
+                makeSwapErrorRight({
+                  message: 'makeSwapperAxiosServiceMonadic: no data was returned',
+                  cause: result,
+                  code: TradeQuoteError.QueryFailed,
+                }),
+              )
+
+            return Ok(result)
+          })
+      }
+    },
+  })
+
+export const getHopByIndex = (
+  quote: TradeQuote | undefined,
+  index: SupportedTradeQuoteStepIndex,
+) => {
+  if (quote === undefined) return undefined
+  if (index > 1) {
+    throw new Error("Index out of bounds - Swapper doesn't currently support more than 2 hops.")
+  }
+  const hop = quote.steps[index]
+
+  return hop
+}
+
+export const executeEvmTransaction = (
+  txToSign: EvmTransactionRequest,
+  callbacks: EvmTransactionExecutionProps,
+) => {
+  return callbacks.signAndBroadcastTransaction(txToSign)
+}
+
+export const createDefaultStatusResponse = (buyTxHash?: string) => ({
+  status: TxStatus.Unknown,
+  buyTxHash,
+  message: undefined,
+})
+
+export const checkEvmSwapStatus = async ({
+  txHash,
+  chainId,
+  assertGetEvmChainAdapter,
+}: {
+  txHash: string
+  chainId: ChainId
+  assertGetEvmChainAdapter: (chainId: ChainId) => EvmChainAdapter
+}): Promise<{
+  status: TxStatus
+  buyTxHash: string | undefined
+  message: string | undefined
+}> => {
+  try {
+    const adapter = assertGetEvmChainAdapter(chainId)
+    const tx = await adapter.httpProvider.getTransaction({ txid: txHash })
+    const status = getTxStatus(tx)
+
+    return {
+      status,
+      buyTxHash: txHash,
+      message: undefined,
+    }
+  } catch (e) {
+    console.error(e)
+    return createDefaultStatusResponse(txHash)
+  }
+}
```

Comment:
nitpick: This file looks like my drawer where I throw all the things I don't wan't to see on my desk, we might want to separate the types and utils into smaller chunks

Reply:
We all have one of those drawers 😂

Reply:
Agree, another one of those things which is obvious now its in a sane location - lets clean up in follow up 🙏